nuxeo-mqueues-core/src/main/java/org/nuxeo/lib/core/mqueues/mqueues/kafka/KafkaUtils.java

```diff
@@ -100,62 +109,61 @@ protected static ZkClient createZkClient(String zkServers) {
-        createTopic(topic, partitions, 1);
+    public static List<List<MQPartition>> rangeAssignments(int threads, Map<String, Integer> streams) {
+        PartitionAssignor assignor = new RangeAssignor();
+        return assignments(assignor, threads, streams);
     }
 
-    public void createTopic(String topic, int partitions, int replicationFactor) {
-        log.info("Creating topic: " + topic + ", partitions: " + partitions + ", replications: " + replicationFactor);
-        if (AdminUtils.topicExists(zkUtils, topic)) {
-            String msg = "Can not create Topic already exists: " + topic;
-            log.error(msg);
-            throw new IllegalArgumentException(msg);
-        }
-        AdminUtils.createTopic(zkUtils, topic, partitions, replicationFactor,
-                new Properties(), RackAwareMode.Safe$.MODULE$);
-        try {
-            // TODO: check if it could be replaced by
-            // TestUtils.waitUntilMetadataIsPropagated(zkUtils.getAllBrokersInCluster(), topic, 0, 5000);
-            waitForTopicCreation(topic, Duration.ofSeconds(5));
-        } catch (InterruptedException e) {
-            Thread.currentThread().interrupt();
-            throw new RuntimeException(e);
-        }
-        AdminUtils.deleteAllConsumerGroupInfoForTopicInZK(zkUtils, topic);
+    public static List<List<MQPartition>> roundRobinAssignments(int threads, Map<String, Integer> streams) {
+        PartitionAssignor assignor = new RoundRobinAssignor();
+        return assignments(assignor, threads, streams);
     }
 
-    protected boolean waitForTopicCreation(String topic, Duration timeout) throws InterruptedException {
-        // if you don't wait for a topic to be ready, this raise LEADER_NOT_AVAILABLE warning
-        // and you can expects lots of rebalancing
-        final long timeoutMs = timeout.toMillis();
-        final long deadline = System.currentTimeMillis() + timeoutMs;
-        boolean ret = false;
-        while (!ret && System.currentTimeMillis() < deadline) {
-            ret = allPartitionsAssigned(topic);
-            Thread.sleep(100);
+    protected static List<List<MQPartition>> assignments(PartitionAssignor assignor, int threads,
+            Map<String, Integer> streams) {
+        final List<PartitionInfo> parts = new ArrayList<>();
+        streams.forEach((streamName, size) -> parts.addAll(getPartsFor(streamName, size)));
+        Map<String, PartitionAssignor.Subscription> subscriptions = new HashMap<>();
+        List<String> streamNames = streams.keySet().stream().sorted().collect(Collectors.toList());
+        for (int i = 0; i < threads; i++) {
+            subscriptions.put(String.valueOf(i), new PartitionAssignor.Subscription(streamNames));
         }
-        if (!ret) {
-            log.error("Topic: " + topic + " has some uninitialized partitions.");
+        Cluster cluster = new Cluster("kafka-cluster", Collections.emptyList(), parts, Collections.emptySet(),
```

Comment:
Put `"kafka-cluster"` in a constant.

Reply:
ok this adon is going to be deprecated soon anyway